File: JJangSungWon/[3]Programmers/전화번호 목록.py
def solution(phone_book):
    for i in range(len(phone_book)):
        for j in range(len(phone_book)):
            if i != j:
                cnt = 0
                if len(phone_book[i]) <= len(phone_book[j]):
                    for k in range(len(phone_book[i])):
                        if phone_book[i][cnt] == phone_book[j][k]:
                            cnt += 1
                        else:
                            cnt = 0

                        if cnt == len(phone_book[i]):
                            return False
    return True


if __name__ == "__main__":
    print(solution(["119", "97674223", "1195524421"]))

File: kimho/5.프로그래머스/전화번호_목록.java
import java.util.*;
class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;

        Arrays.sort(phone_book);
        for(int i=0; i<phone_book.length-1; i++)
            for(int j=i+1; j<phone_book.length; j++)
                if(phone_book[j].startsWith(phone_book[i]))
                    return false;
        return answer;
    }
}